### ISSUE TITLE
TSDAE on T5 results in catastrophic forgetting

### ISSUE THREAD

**Issue by paulpmi** [UKPLab/sentence-transformers#1844]
I've been trying out TSDAE as an alternative to MLM as a pretraining task since I've had more success with it in a previous task. It seems to work well for very few iterations until the loss starts to pick up. This results a complete loss to previous T5 known tasks even with frozen layers. I am checking out if there have been any similar tests before giving up on this pretraining method. It seems to not depend on the dataset.

```
class T5TSDAELoss(nn.Module):
    def __init__(
            self,
            model,
    ):
        super(T5TSDAELoss, self).__init__()
        self.model = model
        self.show_loss = 0

        self.current_losses = 0

    def forward(self, sentence_features: Iterable[Dict[str, Tensor]], labels: Tensor):
        source_features, target_features = tuple(sentence_features)

        target_length = target_features['input_ids'].shape[1]
        decoder_input_ids = target_features['input_ids'].clone()[:, :target_length - 1]
        label_ids = target_features['input_ids'][:, 1:]

        outputs = self.model(
            input_ids=source_features['input_ids'], decoder_input_ids=decoder_input_ids,
            decoder_attention_mask=source_features['attention_mask'][:, 0:1],
            labels=label_ids.reshape(-1)
        )
        if self.show_loss >= 10:
            print(self.current_losses / self.show_loss, flush=True)
            self.show_loss = 0
            self.current_losses = 0
        else:
            self.show_loss += 1

        self.current_losses += outputs.loss
        return outputs.loss
```

the way I've injected the model into the loss
```
model = AutoModelForSeq2SeqLM.from_pretrained(model_name, device_map='auto')
model = SentenceTransformer(modules=[model])
train_loss = T5TSDAELoss(model._modules['0'])
```

the rest is the same as the example in the tsdae example file.
